docs/en/rules/Azure.MySQL.AADOnly.md

```diff
@@ -0,0 +1,89 @@
+---
+severity: Important
+pillar: Security
+category: Identity and access management
+resource: Azure Database for MySQL
+online version: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.MySQL.AADOnly/
+---
+
+# Azure AD-only authentication
+
+## SYNOPSIS
+
+Ensure Azure AD-only authentication is enabled with Azure Database for MySQL databases.
+
+## DESCRIPTION
+
+Azure Database for MySQL supports authentication with MySQL logins and Azure AD authentication.
+
+By default, authentication with MySQL logins is enabled.
+MySQL logins are unable to provide sufficient protection for identities.
+Azure AD authentication provides strong protection controls including conditional access, identity governance, and privileged identity management.
+
+Once you decide to use Azure AD authentication, you can disable authentication with MySQL logins.
+
+Azure AD-only authentication is only supported for the flexible server deployment model with MySQL 5.7 and newer.
+
+## RECOMMENDATION
+
+Consider using Azure AD-only authentication.
+Also consider using Azure Policy for Azure AD-only authentication with Azure Database for MySQL.
+
+## EXAMPLES
+
+### Configure with Azure template
+
+To deploy Azure Database for MySQL flexible servers that pass this rule:
+
+- Configure the `Microsoft.DBforMySQL/flexibleServers/configurations` sub-resource.
+- Set the `name` to `aad_auth_only`.
+- Set the `properties.currentValue` to `ON`.
```

Comment:
Prop `currentValue` is related to reading the value we should use `value`.
Also I think `source` is required.

```suggestion
- Set the `properties.value` to `ON`.
- Set the `properties.source` to `user-override`.
```